extensions/ql-vscode/src/view/method-modeling/MultipleModeledMethodsPanel.tsx

```diff
@@ -60,6 +60,14 @@ export const MultipleModeledMethodsPanel = ({
 }: MultipleModeledMethodsPanelProps) => {
   const [selectedIndex, setSelectedIndex] = useState<number>(0);
 
+  useEffect(() => {
```

Comment:
Hmm, this change actually breaks adding a new model, because in `handleAddClick` we try to set `selectedIndex` but it fails because the number of models hasn't been updated yet. The list of models needs to get sent to the extension host, be updated in the modeling store, and then get sent back.

I'm not really sure how to best solve this. I can think of a couple of non-ideal solutions:
- One option is the `MultipleModeledMethodsPanel` stores the list of modeled methods as state as well as props. Then it can update its internal state right away. I don't think this is a great solution as it introduces extra complexity and places for things to go wrong.
- Another options is the `MultipleModeledMethodsPanel` could have some state to remember that it just tried to add a model, and then when we see the  set of modeled methods change we update `selectedIndex` then. It's a bit clunky though.

Reply:
If you want to test this, the user visible effect is that when you click "add a new model" you stay on model 1, so you're now viewing model 1/2. The intended behaviour is that you would be viewing model 2/2, i.e. the new unmodeled method you just added.